src/Core/Managed/Net45/Extensibility/Implementation/RichPayloadEventSource.cs

```diff
@@ -0,0 +1,498 @@
+                    }
+
+                case ExceptionTelemetry.TelemetryName:
+                    {
+                        var telemetryItem = item as ExceptionTelemetry;
+                        var data = telemetryItem.Data;
+                        extendedData = new
+                        {
+                            PartA_iKey = telemetryItem.Context.InstrumentationKey,
+                            PartA_Tags = telemetryItem.Context.Tags,
+                            PartB_ExceptionData = new
+                            {
+                                exceptions = data.exceptions.Select(i => new
+                                {
+                                    i.hasFullStack,
+                                    i.id,
+                                    i.message,
+                                    i.outerId,
+                                    parsedStack = i.parsedStack.Select(j => new { j.assembly, j.fileName, j.level, j.line, j.method }),
+                                    i.stack,
+                                    i.typeName
+                                }),
+                                data.handledAt,
+                                data.measurements,
+                                data.problemId,
+                                data.properties,
+                                data.severityLevel,
+                                data.ver
+                            }
+                        };
+                        break;
+                    }
+
+                case PerformanceCounterTelemetry.TelemetryName:
+                    {
+                        var telemetryItem = item as PerformanceCounterTelemetry;
+                        var data = telemetryItem.Data;
+                        extendedData = new
+                        {
+                            PartA_iKey = telemetryItem.Context.InstrumentationKey,
+                            PartA_Tags = telemetryItem.Context.Tags,
+                            PartB_PerformanceCounterData = new { data.categoryName, data.count, data.counterName, data.instanceName, data.kind, data.max, data.min, data.properties, data.stdDev, data.value, data.ver }
+                        };
+                        break;
+                    }
+
+                case PageViewTelemetry.TelemetryName:
+                    {
+                        var telemetryItem = item as PageViewTelemetry;
+                        var data = telemetryItem.Data;
+                        extendedData = new
+                        {
+                            PartA_iKey = telemetryItem.Context.InstrumentationKey,
+                            PartA_Tags = telemetryItem.Context.Tags,
+                            PartB_PageViewPerfData = new { data.duration, data.measurements, data.name, data.properties, data.url, data.ver }
+                        };
+                        break;
+                    }
+
+                case SessionStateTelemetry.TelemetryName:
```

Comment:
nitpick: There are a lot of string compare cases in the switch statement, you could build a jump table instead.